### PR TITLE
[Processing] Calculate min coverage on execution

### PR DESCRIPTION
**Pull request by arnaud-morvan**
Here I move the min coverage calculation from the extent selection widget to the execution method of the algorithm.

IMHO calculated extent should not be save anywhere (nor in batch dialog nor in modeler) but only calculated at execution time, so this make sense.

Note that this removes duplicate code : 
- in ModelerAlgorithm and ExtentSelectionPanel for the extent calculation.
- in panel's setParamValue method for get the extent at last.

Note the stats : +65 −138

Merging this will makes me happing for the widget abstraction I'm working on.

### CODE REVIEW

**Comment by volaya** [comment]
I like it, and it make sense, but I am trying to see if there is a situation where this might be a problem.

Now we have no option for setting a null extent, so it means a extent cannot be optional. Probably not a big problem, since that makes little sense, i think...

I am merging it, anyway. Good work!

File: python/plugins/processing/gui/ExtentSelectionPanel.py [hunk 1]
```diff
@@ -205,10 +152,9 @@ def setValueFromRect(self, r):
         self.dialog.activateWindow()
 
     def getValue(self):
-        if unicode(self.leText.text()).strip() != '':
-            return unicode(self.leText.text())
-        else:
-            return self.getMinCoveringExtent()
+        if unicode(self.leText.text()).strip() == '':
+            return None
```
Comment (volaya) [inline python/plugins/processing/gui/ExtentSelectionPanel.py]: If this is None, how is that then set in the parameter. By looking at https://github.com/qgis/QGIS/blob/master/python/plugins/processing/core/parameters.py#L197, it wil return False, unless the parameter is optional (which is not by default). Can you make the changes in the parameters module to fix that? thanks!

Comment (arnaud-morvan) [inline python/plugins/processing/gui/ExtentSelectionPanel.py]: I think that we should not set ParameterExtent optional by default, as we don't know if there if a ParameterDataObject on extent parameter creation.

If we have a pointer to algorithm (like with the QObject inheritance with parent/child relationship), we could test in ParameterExtent.setValue if we can use auto extent.

Another solution should be to set correct optional value in all algorithms.

Comment (arnaud-morvan) [inline python/plugins/processing/gui/ExtentSelectionPanel.py]: On reflexion optional = True by default is a good workaround for the moment. I will set it soon.
Thanks for review.

**Pull request merged**